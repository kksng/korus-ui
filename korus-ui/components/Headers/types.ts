import * as React from 'react';

export interface HeaderProps extends React.HTMLAttributes<HTMLHeadingElement> {
  [x: string]: unknown,
  ref?: React.Ref<HeaderRefCurrent>,
  shouldRender?: boolean,
}

export interface HeaderRefCurrent {
  wrapper: HTMLHeadingElement | null,
}
