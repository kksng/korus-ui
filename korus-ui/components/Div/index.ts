import * as React from 'react';
import { htmlTagFactory } from '../../src/HTMLTagsFactory';

export interface DivProps extends React.HTMLAttributes<HTMLDivElement> {
  [x: string]: unknown,
  ref?: React.Ref<DivRefCurrent>,
  shouldRender?: boolean,
}

export interface DivRefCurrent {
  wrapper: HTMLDivElement | null,
}

export const Div = htmlTagFactory('Div') as React.FC<DivProps>;
