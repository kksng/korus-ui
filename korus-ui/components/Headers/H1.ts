import * as React from 'react';
import { htmlTagFactory } from '../../src/HTMLTagsFactory';
import { HeaderProps } from './types';

export const H1 = htmlTagFactory('H1') as React.FC<HeaderProps>;
