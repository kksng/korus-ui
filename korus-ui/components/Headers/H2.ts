import * as React from 'react';
import { htmlTagFactory } from '../../src/HTMLTagsFactory';
import { HeaderProps } from './types';

export const H2 = htmlTagFactory('H2') as React.FC<HeaderProps>;
