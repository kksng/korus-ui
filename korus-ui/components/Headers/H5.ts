import * as React from 'react';
import { htmlTagFactory } from '../../src/HTMLTagsFactory';
import { HeaderProps } from './types';

export const H5 = htmlTagFactory('H5') as React.FC<HeaderProps>;
