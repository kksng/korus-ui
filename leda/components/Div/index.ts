import * as React from 'react';
import { htmlTagFactory } from '../../src/HTMLTagsFactory';

export interface DivProps extends React.HTMLAttributes<HTMLDivElement> {
  ref?: React.Ref<DivRefCurrent>,
  shouldRender?: boolean,
  [x: string]: unknown,
}

export interface DivRefCurrent {
  wrapper: HTMLDivElement | null,
}

export const Div = htmlTagFactory('Div') as React.FC<DivProps>;
