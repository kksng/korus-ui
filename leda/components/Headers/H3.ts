import * as React from 'react';
import { htmlTagFactory } from '../../src/HTMLTagsFactory';
import { HeaderProps } from './types';

export const H3 = htmlTagFactory('H3') as React.FC<HeaderProps>;
