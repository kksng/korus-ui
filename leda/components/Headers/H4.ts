import * as React from 'react';
import { htmlTagFactory } from '../../src/HTMLTagsFactory';
import { HeaderProps } from './types';

export const H4 = htmlTagFactory('H4') as React.FC<HeaderProps>;
