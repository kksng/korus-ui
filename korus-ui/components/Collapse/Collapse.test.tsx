/* eslint-disable @typescript-eslint/explicit-function-return-type */
import React from 'react';
import { render } from '@testing-library/react';
import { Collapse } from './index';
import { Div } from '../Div';
import { H1 } from '../Headers';

// todo mock css transitions

jest.useFakeTimers();

const MockComponent = () => {
  const [panels, setPanels] = React.useState<string | string[] | null>(['1', '2']);

  return (
    <Collapse
      onSelect={(event) => setPanels(event.component.value)}
      activePanelKey={panels}
    >
      <Collapse.Panel panelKey="1">
        <Collapse.Heading>
          <H1>HEADING 1</H1>
        </Collapse.Heading>
        <Collapse.Body>
          <Div className="test-1">SOME FIRST TEST</Div>
        </Collapse.Body>
      </Collapse.Panel>
      <Collapse.Panel panelKey="2">
        <Collapse.Heading>
          <H1>HEADING 2</H1>
        </Collapse.Heading>
        <Collapse.Body>
          <Div className="test-2">SOME SECOND TEST</Div>
        </Collapse.Body>
      </Collapse.Panel>
    </Collapse>
  );
};

describe('Collapse SNAPSHOTS', () => {
  it('should render closed state', () => {
    const component = (
      <Collapse onSelect={jest.fn()} activePanelKey="0">
        <Collapse.Panel panelKey="1">
          <Collapse.Heading>
            <H1>HEADING</H1>
          </Collapse.Heading>
          <Collapse.Body>
            <Div id="test">SOME TEST</Div>
          </Collapse.Body>
        </Collapse.Panel>
      </Collapse>
    );

    const wrapper = render(component);

    const body = document.getElementById('test')?.parentElement;

    expect(body?.getAttribute('aria-expanded')).toEqual('false');

    expect(wrapper.container).toMatchSnapshot();
  });

  it('should render opened state', () => {
    const component = (
      <Collapse onSelect={jest.fn()} activePanelKey="1">
        <Collapse.Panel panelKey="1">
          <Collapse.Heading>
            <H1>HEADING</H1>
          </Collapse.Heading>
          <Collapse.Body>
            <Div>SOME TEST</Div>
          </Collapse.Body>
        </Collapse.Panel>
      </Collapse>
    );

    const wrapper = render(component);

    expect(wrapper.queryByText('SOME TEST')).toBeInTheDocument();

    expect(wrapper.container).toMatchSnapshot();
  });

  describe('multi-opened state in controlled mode', () => {
    it('should render both bodies opened', () => {
      const wrapper = render(<MockComponent />);

      const one = wrapper.getByText('SOME FIRST TEST');

      expect(one.parentElement?.getAttribute('aria-expanded')).toEqual('true');

      const two = wrapper.getByText('SOME SECOND TEST');

      expect(two.parentElement?.getAttribute('aria-expanded')).toEqual('true');

      expect(wrapper.container).toMatchSnapshot();

      wrapper.unmount();
    });

    it('should render only first body opened', () => {
      const wrapper = render(<MockComponent />);

      wrapper.getByText('HEADING 2').click();

      const one = wrapper.getByText('SOME FIRST TEST');

      expect(one.parentElement?.getAttribute('aria-expanded')).toEqual('true');

      const two = wrapper.getByText('SOME SECOND TEST');

      expect(two.parentElement?.getAttribute('aria-expanded')).toEqual('false');

      expect(wrapper.container).toMatchSnapshot();

      wrapper.unmount();
    });
  });

  describe('accordion state in uncontrolled mode', () => {
    const component = (
      <Collapse isAccordion>
        <Collapse.Panel panelKey="1">
          <Collapse.Heading className="head-1">
            <H1>HEADING 1</H1>
          </Collapse.Heading>
          <Collapse.Body>
            <Div className="test-1">SOME FIRST TEST</Div>
          </Collapse.Body>
        </Collapse.Panel>
        <Collapse.Panel panelKey="2">
          <Collapse.Heading className="head-2">
            <H1>HEADING 2</H1>
          </Collapse.Heading>
          <Collapse.Body>
            <Div className="test-2">SOME SECOND TEST</Div>
          </Collapse.Body>
        </Collapse.Panel>
      </Collapse>
    );

    it('should render both bodies closed', () => {
      const wrapper = render(component);

      const one = wrapper.getByText('SOME FIRST TEST');

      expect(one.parentElement?.getAttribute('aria-expanded')).toEqual('false');

      const two = wrapper.getByText('SOME SECOND TEST');

      expect(two.parentElement?.getAttribute('aria-expanded')).toEqual('false');

      expect(wrapper.container).toMatchSnapshot();

      wrapper.unmount();
    });

    it('should render only first body opened', () => {
      const wrapper = render(component);

      wrapper.getByText('HEADING 1').click();

      const one = wrapper.getByText('SOME FIRST TEST');

      expect(one.parentElement?.getAttribute('aria-expanded')).toEqual('true');

      const two = wrapper.getByText('SOME SECOND TEST');

      expect(two.parentElement?.getAttribute('aria-expanded')).toEqual('false');

      expect(wrapper.container).toMatchSnapshot();

      wrapper.unmount();
    });
  });

  it('should render loading state', () => {
    const component = (
      <Collapse onSelect={jest.fn()} activePanelKey="1">
        <Collapse.Panel panelKey="1">
          <Collapse.Heading>
            <H1>HEADING</H1>
          </Collapse.Heading>
          <Collapse.Body isLoading>
            <Div>SOME TEST</Div>
          </Collapse.Body>
        </Collapse.Panel>
      </Collapse>
    );

    const wrapper = render(component);

    expect(document.querySelector('.loader-wrapper')).toBeInTheDocument();

    expect(wrapper.container).toMatchSnapshot();
  });
});

describe('Collapse HANDLERS', () => {
  it('should call onSelect handler when click on Collapse.Heading', () => {
    const onSelectHandler = jest.fn();

    const component = (
      <Collapse onSelect={onSelectHandler}>
        <Collapse.Panel panelKey="1">
          <Collapse.Heading>
            <H1>HEADING</H1>
          </Collapse.Heading>
          <Collapse.Body>
            <Div>SOME TEST</Div>
          </Collapse.Body>
        </Collapse.Panel>
      </Collapse>
    );

    const wrapper = render(component);

    wrapper.getByText('HEADING').click();

    expect(onSelectHandler).toHaveBeenCalled();
  });
});

describe('Collapse ATTRIBUTES', () => {
  it('should be opened if activePanelKey equals panelKey', () => {
    const MyTestingComponent = () => {
      const [panels, setPanels] = React.useState<string | string[] | null>([]);

      return (
        <Collapse isAccordion onSelect={(event) => setPanels(event.component.value)} activePanelKey={panels}>
          <Collapse.Panel panelKey="1">
            <Collapse.Heading>
              <H1>HEADING 1</H1>
            </Collapse.Heading>
            <Collapse.Body>
              <Div className="first">SOME FIRST TEST</Div>
            </Collapse.Body>
          </Collapse.Panel>
          <Collapse.Panel panelKey="2">
            <Collapse.Heading>
              <H1>HEADING 2</H1>
            </Collapse.Heading>
            <Collapse.Body>
              <Div className="second">SOME SECOND TEST</Div>
            </Collapse.Body>
          </Collapse.Panel>
        </Collapse>
      );
    };

    const wrapper = render(<MyTestingComponent />);

    const one = wrapper.getByText('SOME FIRST TEST');

    const two = wrapper.getByText('SOME SECOND TEST');

    // Collapse should be closed when mounted
    expect(one.parentElement?.getAttribute('aria-expanded')).toEqual('false');

    expect(two.parentElement?.getAttribute('aria-expanded')).toEqual('false');

    // Collapse should open first panel if heading was clicked
    wrapper.getByText('HEADING 1').click();

    expect(one.parentElement?.getAttribute('aria-expanded')).toEqual('true');

    // Collapse should close previous panel and open different if its heading was clicked
    wrapper.getByText('HEADING 2').click();

    expect(one.parentElement?.getAttribute('aria-expanded')).toEqual('false');

    expect(two.parentElement?.getAttribute('aria-expanded')).toEqual('true');
  });

  describe('wrapper', () => {
    describe('usage Div as a wrapper', () => {
      it('should wrap Collapse.Heading', () => {
        const component = (
          <Collapse>
            <Collapse.Panel panelKey="1">
              <Collapse.Heading wrapperRender={() => <Div className="test" _box _active><span>SHOOT MYSELF IN THE LEG!</span></Div>}>
                <H1>HEADING 1</H1>
              </Collapse.Heading>
              <Collapse.Body>
                <Div className="first">SOME FIRST TEST</Div>
              </Collapse.Body>
            </Collapse.Panel>
          </Collapse>
        );

        const wrapper = render(component);

        expect(wrapper.queryByText('SHOOT MYSELF IN THE LEG!')).toBeInTheDocument();
      });

      it('should wrap Collapse.Body', () => {
        const component = (
          <Collapse onSelect={jest.fn()} activePanelKey="1">
            <Collapse.Panel panelKey="1">
              <Collapse.Heading>
                <H1>HEADING 1</H1>
              </Collapse.Heading>
              <Collapse.Body wrapperRender={() => <Div className="test" _box _active><span>SHOOT MYSELF IN THE LEG!</span></Div>}>
                <Div className="first">SOME FIRST TEST</Div>
              </Collapse.Body>
            </Collapse.Panel>
          </Collapse>
        );

        const wrapper = render(component);

        expect(wrapper.queryByText('SHOOT MYSELF IN THE LEG!')).toBeInTheDocument();

        expect(wrapper.queryByText('SOME FIRST TEST')).not.toBeInTheDocument();
      });

      it('should wrap Collapse.Panel', () => {
        const component = (
          <Collapse onSelect={jest.fn()}>
            <Collapse.Panel panelKey="1" wrapperRender={() => <Div _box _active><span>SHOOT MYSELF IN THE LEG!</span></Div>}>
              <Div>TEST</Div>
            </Collapse.Panel>
          </Collapse>
        );

        const wrapper = render(component);

        expect(wrapper.queryByText('SHOOT MYSELF IN THE LEG!')).toBeInTheDocument();

        expect(wrapper.queryByText('TEST')).not.toBeInTheDocument();
      });
    });

    describe('usage span as a wrapper', () => {
      it('should wrap Collapse.Heading', () => {
        const component = (
          <Collapse onSelect={jest.fn()}>
            <Collapse.Panel panelKey="1">
              <Collapse.Heading wrapperRender={() => <span className="box active"><p>SHOOT MYSELF IN THE LEG!</p></span>}>
                <H1>HEADING 1</H1>
              </Collapse.Heading>
              <Collapse.Body>
                <Div className="first">SOME FIRST TEST</Div>
              </Collapse.Body>
            </Collapse.Panel>
          </Collapse>
        );

        const wrapper = render(component);

        expect(wrapper.queryByText('SHOOT MYSELF IN THE LEG!')).toBeInTheDocument();

        expect(wrapper.queryByText('SOME FIRST TEST')).toBeInTheDocument();

        expect(wrapper.queryByText('HEADING 1')).not.toBeInTheDocument();
      });

      it('should wrap Collapse.Body', () => {
        const component = (
          <Collapse onSelect={jest.fn()} activePanelKey="1">
            <Collapse.Panel panelKey="1">
              <Collapse.Heading>
                <H1>HEADING 1</H1>
              </Collapse.Heading>
              <Collapse.Body wrapperRender={() => <span className="box active"><p>SHOOT MYSELF IN THE LEG!</p></span>}>
                <Div className="first">SOME FIRST TEST</Div>
              </Collapse.Body>
            </Collapse.Panel>
          </Collapse>
        );

        const wrapper = render(component);

        expect(wrapper.queryByText('HEADING 1')).toBeInTheDocument();

        expect(wrapper.queryByText('SHOOT MYSELF IN THE LEG!')).toBeInTheDocument();

        expect(wrapper.queryByText('SOME FIRST TEST')).not.toBeInTheDocument();
      });

      it('should wrap Collapse.Panel', () => {
        const component = (
          <Collapse onSelect={jest.fn()}>
            <Collapse.Panel panelKey="1" wrapperRender={() => <span className="box active"><p>SHOOT MYSELF IN THE LEG!</p></span>}>
              <Div>TEST</Div>
            </Collapse.Panel>
          </Collapse>
        );

        const wrapper = render(component);

        expect(wrapper.queryByText('SHOOT MYSELF IN THE LEG!')).toBeInTheDocument();

        expect(wrapper.queryByText('TEST')).not.toBeInTheDocument();
      });
    });
    describe('usage span as a wrapper without children', () => {
      it('should wrap Collapse.Heading', () => {
        const component = (
          <Collapse onSelect={jest.fn()}>
            <Collapse.Panel panelKey="1">
              <Collapse.Heading wrapperRender={() => <span className="box active" />}>
                <H1>HEADING 1</H1>
              </Collapse.Heading>
              <Collapse.Body>
                <Div className="first">SOME FIRST TEST</Div>
              </Collapse.Body>
            </Collapse.Panel>
          </Collapse>
        );

        const wrapper = render(component);

        expect(document.querySelector('.box.active')).toBeInTheDocument();

        expect(wrapper.queryByText('SOME FIRST TEST')).toBeInTheDocument();

        expect(wrapper.queryByText('HEADING 1')).not.toBeInTheDocument();
      });

      it('should wrap Collapse.Body', () => {
        const component = (
          <Collapse onSelect={jest.fn()} activePanelKey="1">
            <Collapse.Panel panelKey="1">
              <Collapse.Heading>
                <H1>HEADING 1</H1>
              </Collapse.Heading>
              <Collapse.Body wrapperRender={() => <span className="box active" />}>
                <Div className="first">SOME FIRST TEST</Div>
              </Collapse.Body>
            </Collapse.Panel>
          </Collapse>
        );

        const wrapper = render(component);

        expect(document.querySelector('.box.active')).toBeInTheDocument();

        expect(wrapper.queryByText('SOME FIRST TEST')).not.toBeInTheDocument();
      });

      it('should wrap Collapse.Panel', () => {
        const component = (
          <Collapse onSelect={jest.fn()}>
            <Collapse.Panel panelKey="1" wrapperRender={() => <span className="box active" />}>
              <Div>TEST</Div>
            </Collapse.Panel>
          </Collapse>
        );

        const wrapper = render(component);

        expect(document.querySelector('.box.active')).toBeInTheDocument();

        expect(wrapper.queryByText('TEST')).not.toBeInTheDocument();
      });
    });
  });
});
