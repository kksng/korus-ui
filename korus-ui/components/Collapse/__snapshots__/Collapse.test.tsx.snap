// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapse SNAPSHOTS accordion state in uncontrolled mode should render both bodies closed 1`] = `
<div>
  <div
    class="collapse-heading-wrapper head-1"
  >
    <h1>
      HEADING 1
    </h1>
    <span
      class="collapse-heading-icon"
    />
  </div>
  <div
    aria-expanded="false"
    style="transition: height 250ms cubic-bezier(.4, 0, .2, 1); will-change: height; height: 0px; overflow: hidden; visibility: hidden;"
  >
    <div
      class="test-1"
    >
      SOME FIRST TEST
    </div>
  </div>
  <div
    class="collapse-heading-wrapper head-2"
  >
    <h1>
      HEADING 2
    </h1>
    <span
      class="collapse-heading-icon"
    />
  </div>
  <div
    aria-expanded="false"
    style="transition: height 250ms cubic-bezier(.4, 0, .2, 1); will-change: height; height: 0px; overflow: hidden; visibility: hidden;"
  >
    <div
      class="test-2"
    >
      SOME SECOND TEST
    </div>
  </div>
</div>
`;

exports[`Collapse SNAPSHOTS accordion state in uncontrolled mode should render only first body opened 1`] = `
<div>
  <div
    class="collapse-heading-wrapper head-1"
  >
    <h1>
      HEADING 1
    </h1>
    <span
      class="collapse-heading-icon icon-expanded"
    />
  </div>
  <div
    aria-expanded="true"
    style="transition: height 250ms cubic-bezier(.4, 0, .2, 1); will-change: height; height: 0px; overflow: hidden; visibility: hidden;"
  >
    <div
      class="test-1"
    >
      SOME FIRST TEST
    </div>
  </div>
  <div
    class="collapse-heading-wrapper head-2"
  >
    <h1>
      HEADING 2
    </h1>
    <span
      class="collapse-heading-icon"
    />
  </div>
  <div
    aria-expanded="false"
    style="transition: height 250ms cubic-bezier(.4, 0, .2, 1); will-change: height; height: 0px; overflow: hidden; visibility: hidden;"
  >
    <div
      class="test-2"
    >
      SOME SECOND TEST
    </div>
  </div>
</div>
`;

exports[`Collapse SNAPSHOTS multi-opened state in controlled mode should render both bodies opened 1`] = `
<div>
  <div
    class="collapse-heading-wrapper"
  >
    <h1>
      HEADING 1
    </h1>
    <span
      class="collapse-heading-icon icon-expanded"
    />
  </div>
  <div
    aria-expanded="true"
    style="transition: height 250ms cubic-bezier(.4, 0, .2, 1); will-change: height; height: auto; overflow: visible; visibility: visible;"
  >
    <div
      class="test-1"
    >
      SOME FIRST TEST
    </div>
  </div>
  <div
    class="collapse-heading-wrapper"
  >
    <h1>
      HEADING 2
    </h1>
    <span
      class="collapse-heading-icon icon-expanded"
    />
  </div>
  <div
    aria-expanded="true"
    style="transition: height 250ms cubic-bezier(.4, 0, .2, 1); will-change: height; height: auto; overflow: visible; visibility: visible;"
  >
    <div
      class="test-2"
    >
      SOME SECOND TEST
    </div>
  </div>
</div>
`;

exports[`Collapse SNAPSHOTS multi-opened state in controlled mode should render only first body opened 1`] = `
<div>
  <div
    class="collapse-heading-wrapper"
  >
    <h1>
      HEADING 1
    </h1>
    <span
      class="collapse-heading-icon icon-expanded"
    />
  </div>
  <div
    aria-expanded="true"
    style="transition: height 250ms cubic-bezier(.4, 0, .2, 1); will-change: height; height: auto; overflow: visible; visibility: visible;"
  >
    <div
      class="test-1"
    >
      SOME FIRST TEST
    </div>
  </div>
  <div
    class="collapse-heading-wrapper"
  >
    <h1>
      HEADING 2
    </h1>
    <span
      class="collapse-heading-icon"
    />
  </div>
  <div
    aria-expanded="false"
    style="transition: height 250ms cubic-bezier(.4, 0, .2, 1); will-change: height; height: auto; overflow: visible; visibility: visible;"
  >
    <div
      class="test-2"
    >
      SOME SECOND TEST
    </div>
  </div>
</div>
`;

exports[`Collapse SNAPSHOTS should render closed state 1`] = `
<div>
  <div
    class="collapse-heading-wrapper"
  >
    <h1>
      HEADING
    </h1>
    <span
      class="collapse-heading-icon"
    />
  </div>
  <div
    aria-expanded="false"
    style="transition: height 250ms cubic-bezier(.4, 0, .2, 1); will-change: height; height: 0px; overflow: hidden; visibility: hidden;"
  >
    <div
      id="test"
    >
      SOME TEST
    </div>
  </div>
</div>
`;

exports[`Collapse SNAPSHOTS should render loading state 1`] = `
<div>
  <div
    class="collapse-heading-wrapper"
  >
    <h1>
      HEADING
    </h1>
    <span
      class="collapse-heading-icon icon-expanded"
    />
  </div>
  <div
    aria-expanded="true"
    style="transition: height 250ms cubic-bezier(.4, 0, .2, 1); will-change: height; height: auto; overflow: visible; visibility: visible;"
  >
    <div
      class="loader-wrapper"
    >
      <div
        class="loader-container"
      >
        <span
          class="loader-element"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Collapse SNAPSHOTS should render opened state 1`] = `
<div>
  <div
    class="collapse-heading-wrapper"
  >
    <h1>
      HEADING
    </h1>
    <span
      class="collapse-heading-icon icon-expanded"
    />
  </div>
  <div
    aria-expanded="true"
    style="transition: height 250ms cubic-bezier(.4, 0, .2, 1); will-change: height; height: auto; overflow: visible; visibility: visible;"
  >
    <div>
      SOME TEST
    </div>
  </div>
</div>
`;
